### PR TITLE
CADC-11992 - refactor success and failure reporting.

### PR DESCRIPTION
**Pull request by SharonGoliath**

### CODE REVIEW

File: caom2pipe/tests/test_reporting.py [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+                    file_info_side_effect = store_modified_true_file_info_get_side_effect
+                    skipped_sum = store_modified_true_skipped_sum
+                if not clean_up:
+                    put_side_effect = clean_up_false_put_side_effect
+                    skipped_sum = store_modified_false_skipped_sum
+                test_config.cleanup_files_when_storing = clean_up
+                test_config.store_modified_files_only = store_modified
+
+                # for using_local in [True, False]:
+                for using_local in [True]:
```
Comment (andamian) [inline caom2pipe/tests/test_reporting.py:179]: `False` is missing from the list

File: caom2pipe/tests/test_reporting.py [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+        test_config.failure_log_file_name = 'fail.txt'
+        test_config.retry_file_name = 'do_again.txt'
+        test_config.success_log_file_name = 'good.txt'
+        test_config.progress_file_name = 'progress.txt'
+        test_config.rejected_file_name = 'reject.txt'
+        test_config._report_fqn = f'{temp_dir}/report.txt'
+
+        for clean_up in [True, False]:
+            for store_modified in [True, False]:
+                move_calls = store_modified_false_move_calls
```
Comment (andamian) [inline caom2pipe/tests/test_reporting.py:161]: It's a bit hard to follow the logic with these loops that use variables initiated earlier.

File: caom2pipe/tests/test_run_composable.py [hunk 3]
```diff
@@ -1128,15 +1050,23 @@ def test_run_ingest(
         test_config.collection = 'CFHT'
         test_config.proxy_file_name = 'cadcproxy.pem'
         test_config.proxy_fqn = f'{tmp_dir_name}/cadcproxy.pem'
-        test_config.features.supports_latest_client = True
         test_config.use_local_files = False
+        test_config.log_file_directory = tmp_dir_name
```
Comment (andamian) [inline caom2pipe/tests/test_run_composable.py:1054]: This seems to repeat throughout the file. Could it be factored out in a separate function?

File: caom2pipe/transfer_composable.py [hunk 4]
```diff
@@ -95,15 +95,7 @@ class Transfer:
...
-        self._observable = None
-
-    @property
-    def observable(self):
-        return self._observable
-
-    @observable.setter
-    def observable(self, value):
-        self._observable = value
+        self._logger = logging.getLogger(self.__class__.__name__)
```
Comment (andamian) [inline caom2pipe/transfer_composable.py:98]: Can't the logger be defined at the module (`transfer_composable`) level?

**Review by andamian — APPROVED** [review]

File: caom2pipe/tests/test_reporting.py (hunk 2 above)
Comment (SharonGoliath) [reply caom2pipe/tests/test_reporting.py:161]: The other choice is to write multiple tests with lots of only-mostly duplicate code. Is that a better approach? I could be missing something when it comes to function-level fixtures as a solution for this kind of integration test.

File: caom2pipe/tests/test_run_composable.py (hunk 3 above)
Comment (SharonGoliath) [reply caom2pipe/tests/test_run_composable.py:1054]: I added the method `change_working_directory` to the `manage_composable.Config` class implementation, and shifted to using it along with the `tmpdir` fixture. Does that address your concerns?

File: caom2pipe/transfer_composable.py (hunk 4 above)
Comment (SharonGoliath) [reply caom2pipe/transfer_composable.py:98]: My personal preference is to have the `class` name in the logging output, but if there's a more Pythonic way to declare loggers, point me to a description.

Comment (andamian) [reply caom2pipe/transfer_composable.py:98]: OK

File: caom2pipe/tests/test_reporting.py (hunk 2 above)
Comment (SharonGoliath) [reply caom2pipe/tests/test_reporting.py]: Reorganized this as much as it seem non-repetitive to do. Let me know if it's better.
